extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -396,6 +419,208 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+
+		return this.createGroup(localizedConstants.DatabaseFilesText, [this.databaseFilesTable, databaseFilesButtonContainer], true);
+	}
+
+	/**
+	 * Converts the file object to a data view object
+	 * @param file database file object
+	 * @returns data view object
+	 */
+	private convertToDataView(file: DatabaseFile): any[] {
+		return [
+			file.name,
+			file.type,
+			file.fileGroup,
+			file.sizeInMb,
+			file.isAutoGrowthEnabled ? localizedConstants.AutoGrowthValueStringGenerator(file.type !== this.viewInfo.fileTypesOptions[2]
+				, file.autoFileGrowth.toString()
+				, file.autoFileGrowthType === localizedConstants.PercentText
+				, file.maxSizeLimitInMb) : localizedConstants.NoneText,
+			file.path,
+			file.fileNameWithExtension
+		];
+	}
+
+	private async onAddDatabaseFilesButtonClicked(button: azdata.ButtonComponent): Promise<void> {
+		// Open file dialog to create file
+		const result = await this.openDatabaseFileDialog(button);
+		if (!isUndefinedOrNull(result)) {
+			this.objectInfo.files?.push(result);
+			var newData = this.objectInfo.files?.map(file => {
+				return this.convertToDataView(file);
+			});
+			await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+		}
+	}
+
+	private async onEditDatabaseFilesButtonClicked(button: azdata.ButtonComponent): Promise<void> {
+		if (this.databaseFilesTable.selectedRows.length === 1) {
+			const result = await this.openDatabaseFileDialog(button);
+			if (!isUndefinedOrNull(result)) {
+				this.objectInfo.files[this.databaseFilesTable.selectedRows[0]] = result;
+				var newData = this.objectInfo.files?.map(file => {
+					return this.convertToDataView(file);
+				});
+				await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+			}
+		}
+	}
+
+	/**
+	 * Removes the selected database file from the table
+	 */
+	private async onRemoveDatabaseFilesButtonClicked(): Promise<void> {
+		if (this.databaseFilesTable.selectedRows.length === 1) {
+			this.objectInfo.files?.splice(this.databaseFilesTable.selectedRows[0], 1);
+			var newData = this.objectInfo.files?.map(file => {
+				return this.convertToDataView(file);
+			});
+			await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+		}
+	}
+
+	/**
+	 * Validate the selected row to enable/disable the remove button
+	 * @returns true if the remove button should be enabled, false otherwise
+	 */
+	protected override removeButtonOnRowSelected(): boolean {
+		let isEnabled = true;
+		const selectedRowId = this.objectInfo.files[this.databaseFilesTable.selectedRows[0]].id;
+		// Cannot delete a row file if the Id is 1.
+		if (this.databaseFilesTable.selectedRows.length === 1 && selectedRowId === 1) {
+			isEnabled = false;
+		}
+		// Cannot remove a log file if there are no other log files
```

Comment:
Not sure how this comment relates to the check below..how is checking the type here doing that?